Error TS2451: Cannot redeclare block-scoped variable 'MonacoEnvironment'

When importing the MonacoEditorModule in AppModule in a CLI-generated Angular 10 application, I get the following error: 
```
ERROR in node_modules/monaco-editor/monaco.d.ts:11:13 - error TS2451: Cannot redeclare block-scoped variable 'MonacoEnvironment'.

11 declare let MonacoEnvironment: monaco.Environment | undefined;
               ~~~~~~~~~~~~~~~~~

  node_modules/monaco-editor/esm/vs/editor/editor.api.d.ts:12:9
    12     let MonacoEnvironment: Environment | undefined;
               ~~~~~~~~~~~~~~~~~
    'MonacoEnvironment' was also declared here.
```

[Here](https://github.com/microsoft/monaco-editor/issues/2219) the monaco team mention that probably what's happening is that I am trying to simultaneously import ESM and AMD module, but it seems I'm doing nothing special, just:
```ts
// app.module.ts
import { MonacoEditorModule } from 'ng-monaco-editor/public-api';
//...
  imports: [
    BrowserModule,
    AppRoutingModule,
    MonacoEditorModule
  ],
```
I tried the different approaches with `dynamic-import` and `baseUrl`  but to no avail. Am I missing something obvious? I will happily update the README if this is the case.

--- a/src/monaco-common-editor.component.ts
+++ b/src/monaco-common-editor.component.ts
@@ -17,6 +17,7 @@ import {
 import { ControlValueAccessor } from '@angular/forms';
 import { ResizeSensor, ResizeSensorCallback } from 'css-element-queries';
 import { debounce, isEqual } from 'lodash-es';
+import * as monacoEditor from 'monaco-editor/esm/vs/editor/editor.api';
 
 import {
   MonacoEditor,
@@ -41,8 +42,8 @@ export abstract class MonacoCommonEditorComponent
     OnDestroy,
     DoCheck,
     ControlValueAccessor {
-  private _rootEditor: monaco.editor.IEditor;
-  protected model: monaco.editor.IModel;
+  private _rootEditor: monacoEditor.editor.IEditor;
+  protected model: monacoEditor.editor.IModel;
   protected _value = '';
   protected _prevOptions: MonacoEditorOptions;
   protected destroyed = false;
@@ -57,7 +58,7 @@ export abstract class MonacoCommonEditorComponent
   protected editor: MonacoEditor;
   private relayoutFunction: ResizeSensorCallback;
   private resizeSensorInstance: ResizeSensor;
-  private disposables: monaco.IDisposable[] = [];
+  private disposables: monacoEditor.IDisposable[] = [];
 
   monacoLoaded = false;
 
@@ -205,7 +206,7 @@ export abstract class MonacoCommonEditorComponent
     this.onTouched = fn;
   }
 
-  createModel(value: string, uri?: string): monaco.editor.ITextModel {
+  createModel(value: string, uri?: string): monacoEditor.editor.ITextModel {
     const { monaco } = this.monacoProvider;
     return monaco.editor.createModel(
       value,

--- a/src/monaco-diff-editor.component.ts
+++ b/src/monaco-diff-editor.component.ts
@@ -9,6 +9,7 @@ import {
   forwardRef,
 } from '@angular/core';
 import { NG_VALUE_ACCESSOR } from '@angular/forms';
+import * as monacoEditor from 'monaco-editor/esm/vs/editor/editor.api';
 
 import { MonacoCommonEditorComponent } from './monaco-common-editor.component';
 import { MonacoEditor } from './monaco-editor-config';
@@ -37,7 +38,7 @@ export class MonacoDiffEditorComponent
   @Input()
   originalValue: string;
 
-  protected originalModel: monaco.editor.ITextModel;
+  protected originalModel: monacoEditor.editor.ITextModel;
 
   createEditor(): MonacoEditor {
     this.originalModel = this.createModel(this.originalValue);

--- a/src/monaco-editor-config.ts
+++ b/src/monaco-editor-config.ts
@@ -1,9 +1,11 @@
+import * as monacoEditor from 'monaco-editor/esm/vs/editor/editor.api';
+
 /**
  * All common option fields for monaco are merged together for ease of config.
  */
-export type MonacoEditorOptions = monaco.editor.IStandaloneEditorConstructionOptions;
+export type MonacoEditorOptions = monacoEditor.editor.IStandaloneEditorConstructionOptions;
 
-export type MonacoEditor = monaco.editor.IStandaloneCodeEditor;
+export type MonacoEditor = monacoEditor.editor.IStandaloneCodeEditor;
 
 /**
  * Configuration over monaco editor.

--- a/src/monaco-provider.service.ts
+++ b/src/monaco-provider.service.ts
@@ -1,4 +1,5 @@
 import { Injectable } from '@angular/core';
+import * as monacoEditor from 'monaco-editor/esm/vs/editor/editor.api';
 import { BehaviorSubject } from 'rxjs';
 
 import {
@@ -6,7 +7,7 @@ import {
   MonacoEditorConfig,
   MonacoEditorOptions,
 } from './monaco-editor-config';
-import { Monaco } from './typing';
+import { Monaco } from './typings';
 
 export interface Require {
   <T>(deps: string[], callback: (result: T) => void): void;
@@ -45,7 +46,7 @@ export class MonacoProviderService {
     if (this.monacoEditorConfig.dynamicImport) {
       return this.monacoEditorConfig
         .dynamicImport()
-        .then(monaco => (this._monaco = monaco as Monaco));
+        .then(monaco => (this._monaco = monaco));
     }
 
     if (this.monacoEditorConfig.baseUrl != null) {
@@ -79,7 +80,7 @@ export class MonacoProviderService {
   /**
    * Expose global monaco object
    */
-  get monaco() {
+  get monaco(): Monaco {
     return this._monaco || window.monaco;
   }
 
@@ -131,8 +132,8 @@ export class MonacoProviderService {
 
   createDiffEditor(
     domElement: HTMLElement,
-    options?: monaco.editor.IDiffEditorConstructionOptions,
-  ): monaco.editor.IStandaloneDiffEditor {
+    options?: monacoEditor.editor.IDiffEditorConstructionOptions,
+  ): monacoEditor.editor.IStandaloneDiffEditor {
     this.assertMonaco();
 
     const diffOptions = {
@@ -150,7 +151,7 @@ export class MonacoProviderService {
    */
   colorizeElement(
     domElement: HTMLElement,
-    options?: monaco.editor.IColorizerElementOptions,
+    options?: monacoEditor.editor.IColorizerElementOptions,
   ) {
     this.assertMonaco();
 
@@ -165,7 +166,7 @@ export class MonacoProviderService {
    */
   getLanguageExtensionPoint(
     alias: string,
-  ): monaco.languages.ILanguageExtensionPoint {
+  ): monacoEditor.languages.ILanguageExtensionPoint {
     this.assertMonaco();
 
     return this.monaco.languages

--- a/src/public-api.ts
+++ b/src/public-api.ts
@@ -1,6 +1,3 @@
-/// <reference path="../node_modules/monaco-editor/monaco.d.ts" />
-
-export * from './typing';
 export * from './monaco-editor.component';
 export * from './monaco-diff-editor.component';
 export * from './monaco-common-editor.component';

--- a/src/typing.ts
+++ b/src/typing.ts
@@ -1,1 +1,0 @@
-export type Monaco = typeof monaco;

--- a/src/typings.ts
+++ b/src/typings.ts
@@ -1,0 +1,9 @@
+import * as monaco from 'monaco-editor/esm/vs/editor/editor.api';
+
+export type Monaco = typeof monaco;
+
+declare global {
+  interface Window {
+    monaco: Monaco;
+  }
+}

